Extensions/TileMap/pixi-tilemap-helper.js

```diff
@@ -169,6 +168,151 @@
+    tiledData.defs.tilesets.forEach((tileset) => {
+      const texture = tileset.relPath
+        ? getTexture(tileset.relPath, tilemapResourceName)
+        : null;
+      tileSetAtlases[tileset.uid] = { texture, ...tileset };
+    });
+    const selectedLevel = tiledData.levels[levelIndex > -1 ? levelIndex : 0];
+
+    const layers = [];
+    const textureCache = {};
+    selectedLevel.layerInstances.reverse().forEach((ldtkLayer, layerIndex) => {
+      const layerAtlasTextureRelPath = ldtkLayer['__tilesetRelPath'];
+      const gridSize = ldtkLayer['__gridSize'];
+      const type = ldtkLayer['__type'];
+      const tilesetUid = ldtkLayer['__tilesetDefUid'];
+      const autoLayerTiles = ldtkLayer.autoLayerTiles;
+      const gridTiles = ldtkLayer.gridTiles;
+      const ldtkTiles = [...autoLayerTiles, ...gridTiles];
+      const entities = ldtkLayer.entityInstances || [];
+      const layer = {
+        type: type,
+        autoLayerTiles,
+        ldtkTiles,
+        gridTiles,
+        entityInstances: entities,
+        visible: ldtkLayer.visible,
+        opacity: ldtkLayer['__opacity'],
+      };
+      textureCache[layerIndex] = {};
+      const tileSet = tileSetAtlases[tilesetUid];
+
+      ldtkTiles.forEach((generatedTile) => {
+        if (generatedTile.t in textureCache[layerIndex]) return;
+
+        try {
+          const [x, y] = generatedTile.src;
+          const rect = new PIXI.Rectangle(x, y, gridSize, gridSize);
+
+          // @ts-ignore - atlasTexture is never null here.
+          const texture = new PIXI.Texture(tileSet.texture, rect);
+
+          textureCache[layerIndex][generatedTile.t] = texture;
+        } catch (error) {
+          console.error(
+            'An error occurred while creating a PIXI.Texture to be used in a TileMap:',
+            error
+          );
+          textureCache[layerIndex] = null;
+        }
+      });
+
+      layers.push(layer);
+    });
+
+    const selectedLevelBg = selectedLevel.bgRelPath;
+    textureCache.levelBg = {};
+    if (selectedLevelBg) {
+      if (selectedLevelBg in textureCache.levelBg) return;
+      try {
+        const levelBgTexture = getTexture(selectedLevelBg, tilemapResourceName);
+        const rect = new PIXI.Rectangle(0, 0, selectedLevel.pxWid, selectedLevel.pxHei);
+        // // @ts-ignore - atlasTexture is never null here.
+        const texture = new PIXI.Texture(levelBgTexture, rect);
+        textureCache.levelBg[selectedLevelBg] = texture;
+      } catch (error) {
+        console.error(
+          'An error occurred while creating a PIXI.Texture to be used in a TileMap:',
+          error
+        );
+        textureCache.levelBg[selectedLevelBg] = null;
+      }
+    }
+    /** @type {GenericPixiTileMapData} */
+    const tileMapData = {
+      width: 0,
+      height: 0,
+      tileWidth: 0, //not needed offset
+      tileHeight: 0, // not needed
+      atlasTexture, // oops, every layer can have a different one,, cant use this
```

Comment:
Not sure I understand this comment?

Reply:
just a comment I forgot to remove. Left it there while researching the ldtk format. Says that atlasTexture doesnt need to be specified by the user manually, as it is already specified in ldtk